types/src/trusted_state.rs

```diff
@@ -174,25 +170,26 @@ impl TrustedState {
 
             Ok(TrustedStateChange::Epoch {
                 new_state,
-                latest_li,
                 latest_epoch_change_li: epoch_change_li,
-                latest_validator_set: new_validator_set,
             })
         } else {
             // The ValidatorChangeProof is empty, stale, or only gets us into our
             // current epoch. We then try to verify that the latest ledger info
             // is this epoch.
-            self.verifier.verify(latest_li)?;
-
-            let new_state = TrustedState {
-                latest_version: res_version,
-                verifier: self.verifier.clone(),
-            };
-
-            Ok(TrustedStateChange::Version {
-                new_state,
-                latest_li,
-            })
+            // TODO: store and verify the root hash if the version equals once new_trust_any_genesis_WARNING_UNSAFE
```

Comment:
Are you (or who?) gonna follow up on this soon enough? Even so, this is a "// FIXME"

Reply:
I'm working on it